src/app/containers/ConsentBanner/useConsentBanners/index.test.js

```diff
@@ -0,0 +1,289 @@
+      expect(result.current.showCookieBanner).toBe(true);
+      expect(fetch).not.toHaveBeenCalled();
+    });
+
+    it('sets POLICY_COOKIE when it is not set', () => {
+      Cookies.remove(POLICY_COOKIE);
+      cookieSetterSpy.mockClear();
+
+      renderHook(() => useConsentBanners());
+
+      expect(cookieSetterSpy).toHaveBeenCalledWith(POLICY_COOKIE, '000', {
+        expires: 365,
+        domain: '.bbc.com',
+      });
+      expect(cookieSetterSpy).toHaveBeenCalledTimes(1);
+      expect(fetch).not.toHaveBeenCalled();
+    });
+
+    it('does not set POLICY_COOKIE when its already set', () => {
+      cookieSetterSpy.mockClear();
+
+      renderHook(() => useConsentBanners());
+
+      expect(cookieSetterSpy).not.toHaveBeenCalledWith(
+        POLICY_COOKIE,
+        expect.anything(),
+        expect.anything(),
+      );
+      expect(fetch).not.toHaveBeenCalled();
+    });
+
+    it('should return showCookieBanner=TRUE when EXPLICIT_COOKIE is 0 and sets POLICY_COOKIE when cookie is null', () => {
+      Cookies.set(EXPLICIT_COOKIE, '0');
+      Cookies.set(POLICY_COOKIE, null);
+      cookieSetterSpy.mockClear();
+
+      const { result } = renderHook(() => useConsentBanners());
+
+      expect(cookieSetterSpy).toHaveBeenCalledTimes(1);
+      expect(cookieSetterSpy).toHaveBeenCalledWith(POLICY_COOKIE, '000', {
+        expires: 365,
+        domain: '.bbc.com',
+      });
+      expect(result.current.showCookieBanner).toBe(true);
+      expect(result.current.showPrivacyBanner).toBe(false);
+      expect(fetch).not.toHaveBeenCalled();
+    });
+
+    it('sets POLICY_COOKIE without domain restrictions', () => {
+      global.document.domain = 'www.test.bbc.com';
+      Cookies.set(EXPLICIT_COOKIE, '0');
+      Cookies.set(POLICY_COOKIE, null);
+      cookieSetterSpy.mockClear();
+
+      renderHook(() => useConsentBanners());
+
+      expect(cookieSetterSpy).toHaveBeenCalledWith(POLICY_COOKIE, '000', {
+        expires: 365,
+        domain: '.bbc.com',
+      });
+      expect(fetch).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('handlePrivacyBannerAccepted', () => {
+    it('should return showPrivacyBanner=FALSE when handlePrivacyBannerAccepted is triggered', () => {
+      cookieSetterSpy.mockClear();
+
+      const { result } = renderHook(() => useConsentBanners());
+
+      act(() => {
+        result.current.handlePrivacyBannerAccepted();
+      });
+
+      expect(result.current.showPrivacyBanner).toBe(false);
+      expect(result.current.showCookieBanner).toBe(true);
+      expect(cookieSetterSpy).not.toHaveBeenCalled();
+      expect(fetch).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('handlePrivacyBannerRejected', () => {
+    it('should return showCookieBanner=TRUE when handlePrivacyBannerRejected is triggered', () => {
+      cookieSetterSpy.mockClear();
+
+      const { result } = renderHook(() => useConsentBanners());
+
+      act(() => {
+        result.current.handlePrivacyBannerRejected();
```

Comment:
You can't 'reject' the privacy banner, you just click okay and it should record in the privacy cookie that the user saw it

Reply:
ah ok I'll refactor this to reflect that